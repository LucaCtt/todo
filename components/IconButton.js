import React from "react";
import { Button, Icon } from "@ui-kitten/components";

export default IconButton = ({ icon, ...props }) => (
  <Button
    {...props}
    appearance="ghost"
    accessoryLeft={(props) => <Icon {...props} name={icon} />}
  />
);
