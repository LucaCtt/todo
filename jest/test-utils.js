import React from "react";
import { render } from "@testing-library/react-native";
import * as eva from "@eva-design/eva";
import { EvaIconsPack } from "@ui-kitten/eva-icons";
import { ApplicationProvider, IconRegistry } from "@ui-kitten/components";

const TestProvider = ({ children }) => (
  <>
    <IconRegistry icons={EvaIconsPack} />
    <ApplicationProvider mapping={eva.mapping} theme={eva.light}>
      {children}
    </ApplicationProvider>
  </>
);

const customRender = (ui, options) =>
  render(ui, { wrapper: TestProvider, ...options });

// re-export everything
export * from "@testing-library/react-native";

// override render method
export { customRender as render };
