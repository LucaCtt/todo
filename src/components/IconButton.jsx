import React, { memo } from "react";
import { Button, Icon } from "@ui-kitten/components";

/**
 * A button that shows an icon as its content.
 *
 * @property {string} icon - The name of the icon to show.
 */
const IconButton = ({ icon, ...props }) => (
  <Button
    appearance="ghost"
    {...props}
    accessoryLeft={(props) => <Icon {...props} name={icon} testID="icon" />}
  />
);

export default memo(IconButton);
