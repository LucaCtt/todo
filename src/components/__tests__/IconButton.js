import React from "react";
import { render } from "test-utils";

import IconButton from "../IconButton";

describe("<IconButton />", () => {
  test("matches snapshot", () => {
    const { baseElement } = render(<IconButton icon="person" />);
    expect(baseElement).toMatchSnapshot();
  });

  test("has icon", () => {
    const { getByTestId, baseElement } = render(<IconButton icon="person" />);
    expect(baseElement).toContainElement(getByTestId("icon"));
  });
});
