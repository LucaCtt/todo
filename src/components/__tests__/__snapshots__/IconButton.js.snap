// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton /> matches snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={1}
        appearance="ghost"
        size="medium"
        status="primary"
        style={
          Array [
            undefined,
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderRadius": 4,
                "borderWidth": 1,
                "minHeight": 40,
                "minWidth": 40,
                "paddingHorizontal": 10,
                "paddingVertical": 12,
              },
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={20}
            bbWidth={20}
            fill="#3366FF"
            focusable={false}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "height": 20,
                  "marginHorizontal": 10,
                  "tintColor": "#3366FF",
                  "width": 20,
                },
                Object {
                  "opacity": 1,
                },
                Object {
                  "flex": 0,
                  "height": 20,
                  "width": 20,
                },
              ]
            }
            testID="icon"
            vbHeight={24}
            vbWidth={24}
          >
            <RNSVGGroup
              fill={4281558783}
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "fill",
                ]
              }
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            >
              <RNSVGGroup
                fill={4278190080}
                fillOpacity={1}
                fillRule={1}
                font={Object {}}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={Array []}
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              >
                <RNSVGGroup
                  fill={4278190080}
                  fillOpacity={1}
                  fillRule={1}
                  font={Object {}}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={Array []}
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                >
                  <RNSVGRect
                    fill={4278190080}
                    fillOpacity={1}
                    fillRule={1}
                    height="24"
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    opacity={0}
                    originX={0}
                    originY={0}
                    propList={Array []}
                    rotation={0}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth={1}
                    vectorEffect={0}
                    width="24"
                    x={0}
                    y={0}
                  />
                  <RNSVGPath
                    d="M12 11a4 4 0 1 0-4-4 4 4 0 0 0 4 4z"
                    fill={4278190080}
                    fillOpacity={1}
                    fillRule={1}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    opacity={1}
                    originX={0}
                    originY={0}
                    propList={Array []}
                    rotation={0}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth={1}
                    vectorEffect={0}
                    x={0}
                    y={0}
                  />
                  <RNSVGPath
                    d="M18 21a1 1 0 0 0 1-1 7 7 0 0 0-14 0 1 1 0 0 0 1 1z"
                    fill={4278190080}
                    fillOpacity={1}
                    fillRule={1}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    opacity={1}
                    originX={0}
                    originY={0}
                    propList={Array []}
                    rotation={0}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth={1}
                    vectorEffect={0}
                    x={0}
                    y={0}
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;
